client/components/webpack-build-monitor/index.jsx

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable no-console */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { createStore } from 'redux';
+import cx from 'classnames';
+import { startsWith } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Spinner from 'components/spinner';
+
+const CONNECTED = 'CONNECTED';
+const DISCONNECTED = 'DISCONNECTED';
+const NEEDS_RELOAD = 'NEEDS_RELOAD';
+const BUILDING = 'BUILDING';
+const ERROR = 'ERROR';
+const IDLE = 'IDLE';
+
+// Once completed, set the status to idle unless there's an error
+const doneBuilding = state => ( state === ERROR || state === NEEDS_RELOAD ? state : IDLE );
+
+const MESSAGE_STATUS_MAP = {
+	'[HMR] connected': () => CONNECTED,
+	'[WDS] Disconnected!': () => DISCONNECTED,
+	'[HMR] bundle rebuilding': () => BUILDING,
+	'Building CSS…': () => BUILDING,
+	'[HMR] App is up to date.': doneBuilding,
+	'[WDS] Nothing changed.': doneBuilding,
+	'[WDS] App hot update...': doneBuilding,
+	'Reloading CSS: ': doneBuilding,
+	'[WDS] Errors while compiling.': () => ERROR,
+	'[HMR] Cannot find update. Need to do a full reload!': NEEDS_RELOAD,
+};
+
+const isDoneBuilding = msg => startsWith( msg, '[HMR] bundle rebuilt in' );
+const needsReload = msg => startsWith( msg, '[HMR]  - ./client' );
+
+const reducer = ( state = IDLE, { message } ) => {
+	if ( message in MESSAGE_STATUS_MAP ) {
+		return MESSAGE_STATUS_MAP[ message ]( state );
+	} else if ( isDoneBuilding( message ) ) {
+		return doneBuilding( state );
+	} else if ( needsReload( message ) ) {
+		return NEEDS_RELOAD;
+	}
+
+	return state;
+};
+
+const store = createStore( reducer );
+
+const wrapConsole = fn => ( message, ...args ) => {
+	if ( message in MESSAGE_STATUS_MAP || isDoneBuilding( message ) || needsReload( message ) ) {
+		store.dispatch( { type: 'WebpackBuildMessage', message } );
+	}
+	fn.call( window, message, ...args );
+};
+
+console.error = wrapConsole( console.error );
+console.warn = wrapConsole( console.warn );
+console.log = wrapConsole( console.log );
+
+class WebpackBuildMonitor extends React.Component {
+	state = { status: IDLE };
+
+	componentDidMount() {
+		this.unsubscribe = store.subscribe( () => {
+			this.setState( { status: store.getState() } );
```

Comment:
> But I'm not sure that's a practical concern?

I'd argue it is.  Its not just about rendering multiple build-monitors.  Its also if one gets dismounted/remounted from the VDOM.  While I don't see that happening anytime soon (maybe if we start dis/remounting the whole layout.jsx in case of large scale errors), It feels much too unsafe to have a console.anything override in a constructor.